scheduler/src/storage.rs

```diff
@@ -0,0 +1,120 @@
+use std::collections::HashMap;
+
+/// Defining a trait called IStorage that takes a generic type T.
+trait IStorage<T> {
+    fn get(&self, id: &str) -> Option<&T>;
+    fn get_mut(&mut self, id: &str) -> Option<&mut T>;
+    fn update(&mut self, id: &str, value: T);
+    fn delete(&mut self, id: &str);
+    fn get_all(&self) -> &HashMap<String, T>;
+}
+
+/// `Storage` is a generic type that stores a `HashMap` of `String` keys and `T` values.
```

Comment:
Are you planning to actually store the hashmap in some physical non volatile storage (e.g. a disk)?

Reply:
No, only in RAM. The structure is only a wrapper for the hashmap.